neuromation/api/storage.py

```diff
@@ -254,15 +262,16 @@ async def create(self, uri: URL, data: Union[bytes, AsyncIterator[bytes]]) -> No
             resp  # resp.status == 201
 
     async def stat(self, uri: URL) -> FileStatus:
-        url = self._config.storage_url / self._uri_to_path(uri)
+        uri = self._normalize_uri(uri)
+        url = self._config.storage_url / self._uri_to_path(uri, normalized=True)
         url = url.with_query(op="GETFILESTATUS")
         auth = await self._config._api_auth()
 
         request_time = time.time()
         async with self._core.request("GET", url, auth=auth) as resp:
             self._set_time_diff(request_time, resp)
             res = await resp.json()
-            return _file_status_from_api(res["FileStatus"])
+            return _file_status_from_api(uri.parent, res["FileStatus"])
```

Comment:
What happen when `uri` is `URL("storage://clustername")` or `URL("storage://")`?

Reply:
good point. I'll handle it in the upcoming commit.

Reply:
Done.
Surprisingly, the storage server returns the full path for `stat()` call but names only for `ls`.
Seem like a subtle inconsistency for my eyes but I can live with it.

Now `uri.parent` call is not required